Include/internal/pycore_long.h

```diff
@@ -202,6 +202,14 @@ _PyLong_UnsignedDigitCount(const PyLongObject *op)
     return op->long_value.lv_tag >> NON_SIZE_BITS;
 }
 
+static inline int
+_PyLong_CompactSign(const PyLongObject *op)
+{
+    assert(PyLong_Check(op));
+    assert(_PyLong_IsCompact(op));
+    return 1 - (op->long_value.lv_tag & SIGN_MASK);
+}
+
```

Comment:
Shouldn't this be the new implementation of `_PyLong_Sign()`, if `_PyLong_NonCompactSign()` is removed?  This gets rid of a branch in the proposed version of `_PyLong_Sign()`.

Reply:
They sure look identical to me. Maybe Mark has plans and maybe the compiler would optimize this anyway?
```
if (P(x))
  return F(x);
else
  return F(x);
```
could just become `return F(x);`.

Reply:
We want the freedom to implement the "compact" and non-compact forms differently.
They have the same implementation at the moment, but that will change.

`_PyLong_Sign()` is part of the ABI, so we need to retain it. But almost all code using `_PyLong_Sign()` actually wants to know if an int is negative and should be using `_PyLong_IsNegative()`.